hocon/src/test/scala/com/avsystem/commons/hocon/HoconInputTest.scala

```diff
@@ -1,13 +1,30 @@
 package com.avsystem.commons
 package hocon
 
-import java.time.{Duration, Period}
-
 import com.avsystem.commons.serialization.json.JsonStringOutput
 import com.avsystem.commons.serialization.{GenCodecRoundtripTest, Input, Output}
-import com.typesafe.config.{ConfigFactory, ConfigMemorySize, ConfigValue, ConfigValueFactory, ConfigValueType}
+import com.typesafe.config.*
+
+import java.time.{Duration, Period}
+import scala.concurrent.duration.*
+
+object HoconInputTest {
+  case class CustomCodecsClass(
+    duration: FiniteDuration,
+    jDuration: Duration,
+    fileSize: SizeInBytes,
+    embeddedConfig: Config,
+    period: Period,
+    clazz: Class[?],
+    clazzMap: Map[Class[?], String],
+  )
+  object CustomCodecsClass extends DefaultConfigCompanion[CustomCodecsClass]
+}
 
 class HoconInputTest extends GenCodecRoundtripTest {
```

Comment:
Output still needs testing, the codecs you wrote work 2 ways. A roundtrip check wouldn't hurt as I've seen export/import use cases with these.

Reply:
`HoconOutput` isn't implemented properly, there is no code handling all these custom types

```
 //TODO: writeCustom
```

Reply:
Moreover, I don't see methods in typesafe config that allow e.g. convert a `Duration` instance back to the nice String representation

com.typesafe.config.impl.ConfigImpl#fromAnyRef(java.lang.Object, com.typesafe.config.ConfigOrigin, com.typesafe.config.impl.FromMapMode)

Reply:
I added a testcase to com.avsystem.commons.hocon.HoconGenCodecRoundtripTest